### PR TITLE
ARACHNE-2806 fix states that mistakenly ended up under the wrong countries

### PR DESCRIPTION
**Pull request by YaroslavTir**
ARACHNE-2806 
fix states that mistakenly ended up under the wrong countries

### CODE REVIEW

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql [hunk 1]
```diff
@@ -1,11 +1,62 @@
...
+where state_province_id = austria_ids.id;
+
+--1.2 move states to the USA
+update states_provinces sp_austria set
+    country_id = (SELECT max(id) FROM countries WHERE name = 'United States of America (the)')
+where 0=0
+  and name in ('Alabama', 'Alaska', 'Arizona',
+               'Arkansas', 'California', 'Colorado',
+               'Connecticut', 'Delaware', 'Florida')
+  and country_id = (SELECT max(id) FROM countries WHERE name = 'Austria')
```
Comment (wivern) [inline src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: JOIN (same as above)

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql [hunk 2]
```diff
@@ -1,11 +1,62 @@
...
+--2.1 fix state_id and country_id in the users_data table
+update users_data
+set state_province_id = pair_of_austria_and_usa_ids.usa_id,
+    country_id = (SELECT max(id) FROM countries WHERE name = 'United States of America (the)')
+from (select sp_austria.id austria_id,
+             sp_usa.id     usa_id
+      from states_provinces sp_austria
+      join states_provinces sp_usa
+        on trim(upper(sp_austria.name)) = trim(upper(sp_usa.name))
+       where sp_austria.country_id = (SELECT max(id) FROM countries WHERE name = 'Austria')
```
Comment (wivern) [inline src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: JOIN (same as above)

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql [hunk 3]
```diff
@@ -1,11 +1,62 @@
...
-               'Delaware',
-               'Florida');
\ No newline at end of file
+-- The order of the queries is critically important, so please do not change it.
+
+-- Step 1.
+-- find states that related to the wrong country(Austria instead of USA)
+-- 1.1 set USA country for these states in users_data table
+update users_data
+set country_id = (SELECT max(id) FROM countries WHERE name = 'United States of America (the)')
```
Comment (wivern) [inline src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: Aggregate might be expensive operation. Probably `LIMIT 1` is more suitable here, keeping in mind the country name is unique in the table. Could you compare execution plans to find what operation is cheaper?

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql [hunk 4]
```diff
@@ -1,11 +1,62 @@
...
+from (select sp_austria.id
+      from states_provinces sp_austria
+         where 0=0
```
[the remark this answers is not in the document]
Comment (YaroslavTir) [reply src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: fixed

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql [hunk 5]
```diff
@@ -1,11 +1,62 @@
...
+         and name in ('Alabama', 'Alaska', 'Arizona',
+                      'Arkansas', 'California', 'Colorado',
+                      'Connecticut', 'Delaware', 'Florida')
+         and country_id = (SELECT max(id) FROM countries WHERE name = 'Austria')
```
[the remark this answers is not in the document]
Comment (YaroslavTir) [reply src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: fixed

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql (hunk 1 above)
Comment (YaroslavTir) [reply src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: fixed

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql (hunk 2 above)
Comment (YaroslavTir) [reply src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: fixed

File: src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql (hunk 3 above)
Comment (YaroslavTir) [reply src/main/resources/db/migration/V20200311165500__set-correct-country-for-states.sql]: fixed

**Pull request merged**